plugins/source.go

```diff
@@ -114,19 +116,21 @@ func (p *SourcePlugin) Sync(ctx context.Context, logger zerolog.Logger, spec spe
 	if concurrency == 0 {
 		concurrency = defaultConcurrency
 	}
-	logger.Info().Uint64("concurrency", concurrency).Msg("starting fetch")
+	logger.Info().Uint64("concurrency", concurrency).Msg("starting sync")
 	goroutinesSem := semaphore.NewWeighted(helpers.Uint64ToInt64(concurrency))
 	wg := sync.WaitGroup{}
 	summary := schema.SyncSummary{}
 	startTime := time.Now()
-	tableNames, err := p.interpolateAllResources(spec.Tables)
+	tableNames, err := p.listAndValidateTables(spec.Tables, spec.SkipTables)
 	if err != nil {
 		return nil, err
 	}
 
+	logger.Debug().Interface("tables", tableNames).Msg("got table names")
+
 	for _, table := range p.tables {
```

Comment:
If we already validates tables, let's just use `tableName` and do `range _, table := range tableName` so the validation logic is in one place

Reply:
`tableNames` is of type `[]string`, while `p.tables` is of type `schema.Tables`. We can't range over `tableNames`, since need the `schema.Table` here. We could add a lookup for the `schema.Table` object? I played around with it, but it doesn't seem better